src/projection.ts

```diff
@@ -137,6 +135,27 @@ export class AtlasProjection extends BaseAtlasClass<
     }
   }
 
+  async datasetId(): Promise<UUID> {
+    if (this.project_id !== undefined) {
+      return this.project_id;
+    }
+    if (this._project !== undefined) {
+      return this._project.id;
+    }
+
+    const endpoint = `/v1/project/projection/${this.id}/get/dataset_id`;
+
+    this._project_id_promise = (async () => {
+      const { dataset_id } = (await this.apiCall(endpoint, 'GET')) as {
+        dataset_id: UUID;
+      };
+      this.project_id = dataset_id;
+      return dataset_id;
+    })();
```

Comment:
If promise defined return promise

Else promise = apicall.then(({dataset_id}) => dataset_id)

Return promise.

Reply:
Ie this needs a non-async assignment early.

Reply:
And to check the slot before rerunning the fetch.